rfc/rfc-0000-match-params.md

```diff
@@ -0,0 +1,138 @@
+# Meta
+[meta]: #meta
+- Name: Match params in field selector
+- Start Date: 2022-02-14
+- Author(s): @jwntrs
+- Status: Draft
+- RFC Pull Request:
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The [template switching RFC](https://github.com/vmware-tanzu/cartographer/pull/75) introduced the ability to match templates based on fields in the workload. However matching on params is not well defined. It is currently not possible to match on a field in the supply chain, and matching on fields in the workload is complex. You would need to define something as follows:
+
+```yaml
+  selector:
+    matchFields:
+      - key: workload.spec.params[?(@.name=="promotion")].value    #< ===== not so nice
+        operator: In
+        values: ["gitops"]
+```
+
+Lets introduce a top level params key:
+
+```yaml
+  selector:
+    matchFields:
+      - key: params.promotion    #< ===== use something like this instead
+        operator: In
+        values: ["gitops"]
+```
+
+
+# Motivation
+[motivation]: #motivation
+
+We should be able to match against supply chain params in our field selectors.
+
+# What it is
+[what-it-is]: #what-it-is
+
+```yaml
+apiVersion: kontinue.io/v1alpha1
+kind: ClusterSupplyChain
+metadata:
+  name: responsible-ops
+spec:
+  selector:
+    app: web
+  params:
+  - name: promotion       #< ===== params can be set here
+    value: (gitops|regops)
+  resources:
+   ...
+    - name: promote
+      templateRef:
+        kind: ClusterTemplate
+        options:
+        - name: git-promotion
+          selector:
+            matchFields:
+              - key: params.promotion           #< ===== introduce top level `params` that relies on our param hierarchy
+                operator: In
+                values: ["gitops"]
+        - name: registry-promotion
+          selector:
+            matchFields:
+              - key: params.promotion
+                operator: In
+                values: ["regops"]
+
+---
+apiVersion: kontinue.io/v1alpha1
+kind: Workload
+metadata:
+  name: petclinic
+  labels:
+    app: web
+spec:
+  params:
+  - name: promotion           #< ===== params can also be set here
+    value: (gitops|regops)
```

Comment:
In this example, the workload param would be ignored, as the value has been hard coded by the supply-chain author. Switch `value` to `default` to clarify that the workload param is allowed.

```suggestion
    default: (gitops|regops)
```